stubs/redis/redis/client.pyi

```diff
@@ -39,305 +40,276 @@ def parse_slowlog_get(response, **options): ...
         socket_timeout: float | None,
         socket_connect_timeout: float | None,
         socket_keepalive: bool | None,
         socket_keepalive_options: Mapping[str, int | str] | None,
         connection_pool: ConnectionPool | None,
-        unix_socket_path: Text | None,
-        encoding: Text,
-        encoding_errors: Text,
-        charset: Text | None,
+        unix_socket_path: str | None,
+        encoding: str,
+        encoding_errors: str,
+        charset: str | None,
         decode_responses: Literal[True],
-        errors: Text | None = ...,
+        errors: str | None = ...,
         retry_on_timeout: bool = ...,
         ssl: bool = ...,
-        ssl_keyfile: Text | None = ...,
-        ssl_certfile: Text | None = ...,
+        ssl_keyfile: str | None = ...,
+        ssl_certfile: str | None = ...,
         ssl_cert_reqs: str | int | None = ...,
-        ssl_ca_certs: Text | None = ...,
+        ssl_ca_certs: str | None = ...,
         ssl_check_hostname: bool = ...,
         max_connections: int | None = ...,
         single_connection_client: bool = ...,
         health_check_interval: float = ...,
-        client_name: Text | None = ...,
-        username: Text | None = ...,
+        client_name: str | None = ...,
+        username: str | None = ...,
     ) -> Redis[str]: ...
     @overload
     def __new__(
         cls,
-        host: Text = ...,
+        host: str = ...,
         port: int = ...,
         db: int = ...,
-        password: Text | None = ...,
+        password: str | None = ...,
         socket_timeout: float | None = ...,
         socket_connect_timeout: float | None = ...,
         socket_keepalive: bool | None = ...,
         socket_keepalive_options: Mapping[str, int | str] | None = ...,
         connection_pool: ConnectionPool | None = ...,
-        unix_socket_path: Text | None = ...,
-        encoding: Text = ...,
-        encoding_errors: Text = ...,
-        charset: Text | None = ...,
+        unix_socket_path: str | None = ...,
+        encoding: str = ...,
+        encoding_errors: str = ...,
+        charset: str | None = ...,
         *,
         decode_responses: Literal[True],
-        errors: Text | None = ...,
+        errors: str | None = ...,
         retry_on_timeout: bool = ...,
         ssl: bool = ...,
-        ssl_keyfile: Text | None = ...,
-        ssl_certfile: Text | None = ...,
+        ssl_keyfile: str | None = ...,
+        ssl_certfile: str | None = ...,
         ssl_cert_reqs: str | int | None = ...,
-        ssl_ca_certs: Text | None = ...,
+        ssl_ca_certs: str | None = ...,
         ssl_check_hostname: bool = ...,
         max_connections: int | None = ...,
         single_connection_client: bool = ...,
         health_check_interval: float = ...,
-        client_name: Text | None = ...,
-        username: Text | None = ...,
+        client_name: str | None = ...,
+        username: str | None = ...,
     ) -> Redis[str]: ...
     @overload
     def __new__(
```

Comment:
If I understood correctly, the 3 `__new__` overloads are:
- for passing `decode_responses=True`
- for letting `decode_responses` be its default value
- for passing `decode_responses=False`

I have a few questions:
- Why do we need `__new__` in the first place? The implementation only has `__init__`. Do we really need to duplicate it all into `__init__` methods?
- Why can't we use `decode_responses: Literal[False] = ...` to combine two overloads into one? The default value is `False` at runtime.
- Why does `decode_responses` come before `errors` in two of the overloads, but after `errors` in the third one?